api/staged_products_property_collection_helpers.go

```diff
@@ -0,0 +1,142 @@
+	if value, ok := item.Data[fieldName].(string); ok {
+		return value
+	}
+	return ""
+}
+
+func (item updatedPropertyCollectionItem) setFieldValue(fieldName string, value string) {
+	item.Data[fieldName] = value
+}
+
+func parseUpdatedPropertyCollection(updatedProperty interface{}) (updatedPropertyCollection, error) {
+	var collection updatedPropertyCollection
+
+	if updatedPropertyAsMap, ok := updatedProperty.(map[string]interface{}); ok {
+		rawItems := updatedPropertyAsMap["value"]
+		if rawItemSlice, ok := rawItems.([]interface{}); ok {
+			for _, item := range rawItemSlice {
+				if itemMap, ok := item.(map[string]interface{}); ok {
+					collection = append(collection, updatedPropertyCollectionItem{Data: itemMap})
+				} else {
+					return nil, fmt.Errorf("parseUpdatedPropertyCollection: failed to convert %v to map[string]interface{}", item)
+				}
+			}
+		} else {
+			return nil, fmt.Errorf("parseUpdatedPropertyCollection: failed to convert %v to []interface{}", rawItems)
+		}
+	} else {
+		return nil, fmt.Errorf("parseUpdatedPropertyCollection: failed to convert %v to map[string]interface{}", updatedProperty)
+	}
+	return collection, nil
+}
+
+type responsePropertyCollection []responsePropertyCollectionItem
+type responsePropertyCollectionItem struct {
+	Data map[interface{}]interface{}
+}
+
+func parseResponsePropertyCollection(rawItems interface{}) (responsePropertyCollection, error) {
+	var collection responsePropertyCollection
+	if rawItemSlice, ok := rawItems.([]interface{}); ok {
+		for _, item := range rawItemSlice {
+			if itemMap, ok := item.(map[interface{}]interface{}); ok {
+				collection = append(collection, responsePropertyCollectionItem{Data: itemMap})
+			} else {
+				return nil, fmt.Errorf("parseResponsePropertyCollection: failed to convert %v to map[interface{}]interface{}", item)
+			}
+		}
+	} else {
+		return nil, fmt.Errorf("parseResponsePropertyCollection: failed to convert %v to []interface{}", rawItems)
+	}
+	return collection, nil
+}
+
+func (item responsePropertyCollectionItem) getFieldValue(fieldName string) string {
+	if valueObj, ok := item.Data[fieldName].(map[interface{}]interface{}); ok {
+		if value, ok := valueObj["value"].(string); ok {
+			return value
+		}
+	}
+	return ""
+}
+
+//Find and return first string matching regex pattern
+//returns matched <string, true> if found, or <"",false> if not
+func findFirst(strings []string, fieldRegex string) (string, bool) {
+	r, err := regexp.Compile(fieldRegex)
+	if err != nil {
+		panic(fmt.Sprintf("failed to compile regex: %v because:\n%v", fieldRegex, err))
+	}
+	for _, str := range strings {
+		if r.MatchString(str) {
+			return str, true
+		}
+	}
+	return "", false
+}
+
+//Finds logical key field (if exists)
+//returns <fieldName, true> if map contains field that can be considered a logical key
+//returns <"", false> if no logical key found
+func (item responsePropertyCollectionItem) findLogicalKeyField() (string, bool) {
```

Comment:
how this function is used in associateExistingCollectionGUIDs doesn't require the bool return type. Since there's no opportunity for error, it can just return the found string.

Reply:
Whilst true currently; *not* finding a logical key (because there isn't one) is information we will need in the future when implementing _assign GUIDs using some other strategy_ logic.

Since checking a bool return type seems cleaner than checking for an empty string; I'd prefer to leave this as is for now.

I've attempted to capture this requirement in [this test](https://github.com/pivotal-cf/om/pull/502/commits/9e73e245a6467a586e74af526c8e3a94377db414#diff-cb697bc504ae3e69003a9c246d254be0R160)